utils.go

```diff
@@ -147,6 +148,32 @@ func parseDSN(dsn string) (cfg *config, err error) {
 		return nil, errInvalidDSNNoSlash
 	}
 
+	if cfg.interpolateParams && cfg.collation != defaultCollation {
+		// A whitelist of collations which safe to interpolate parameters.
+		// ASCII and latin-1 are safe since they are single byte encoding.
+		// utf-8 is safe since it doesn't conatins ASCII characters in trailing bytes.
+		safeCollations := []string{"ascii_", "latin1_", "utf8_", "utf8mb4_"}
```

Comment:
The blacklist would then be:
"big5_chinese_ci": 1,
"sjis_japanese_ci": 13,
"gbk_chinese_ci": 28,
"big5_bin": 84,
"gb2312_bin": 86,
"gbk_bin": 87,
"sjis_bin": 88,
"cp932_japanese_ci": 95,
"cp932_bin": 96,

You can also directly check the ID instead of the name then